cadc-download-manager/src/main/java/ca/nrc/cadc/dlm/DownloadTupleFormat.java

```diff
@@ -101,83 +107,88 @@ public DownloadTuple parse(String tupleStr) throws DownloadTupleParsingException
             throw new DownloadTupleParsingException("invalid tuple format: mismatched braces ': " + tupleStr);
         }
 
-        if (openBraceCount > 2) {
-            throw new DownloadTupleParsingException("invalid tuple format: too many braces ': " + tupleStr);
+        // either all or nothing
+        if (openBraceCount != 0 && openBraceCount != 3) {
+            throw new DownloadTupleParsingException("invalid tuple format: " + tupleStr);
         }
 
         // Split out the tuple parts
-        String [] tupleParts = tupleStr.split("\\{");
-        String tmpLabel;
-
-        if (tupleParts.length > 3) {
-            throw new DownloadTupleParsingException("tuple has too many parts. expected max 3 and found "
-                + tupleParts.length + ": " + tupleStr);
-        }
-        log.debug("tuple parts count: " + tupleParts.length);
-
-        // Get any label that might be there
-        if (tupleParts.length == 3) {
-            // grab optional third [2] parameter as label
-            String l = tupleParts[2];
-            if (l.length() > 1) {
-                // trim off trailing "}".
-                // guaranteed to be there due to
-                // check for equal occurrences of { and } above.
-                tmpLabel = convertLabelText(l.substring(0, l.length() - 1));
-            } else {
-                // invalid format
-                throw new DownloadTupleParsingException("invalid label format: " + tupleStr);
+        String[] tupleParts = tupleStr.split("\\{");
+        // Get tuple URI - should at least have this.
+        URI tmpURI = null;
+        String uriStr = tupleParts[0];
+        log.debug("uri given: " + uriStr);
+        if (StringUtil.hasLength(uriStr)) {
+            try {
+                tmpURI = new URI(uriStr);
+            } catch (URISyntaxException u) {
+                throw new DownloadTupleParsingException("id parsing error: " + u + ": " + tupleStr);
             }
         } else {
-            tmpLabel = null;
+            // invalid format - has to at least be a single URI passed in
+            throw new DownloadTupleParsingException("zero length id found: " + tupleStr);
         }
 
-        // Get any cutout that might be there
-        Shape tmpShape = null;
-        if (tupleParts.length > 1) {
-            String sd = tupleParts[1];
-            if (sd.length() > 1) {
+        // Create new DownloadTuple
+        DownloadTuple newTuple = new DownloadTuple(tmpURI);
+
+        if (openBraceCount == 3) {
+            String tmpPosStr = tupleParts[1];
+            if (tmpPosStr.length() > 1) {
```

Comment:
StringUtil.hasLength() does this too and is more consistent with everywhere else in this file.

Reply:
done